lib/plugin-compat.php

```diff
@@ -31,4 +31,4 @@ function gutenberg_remove_wpcom_markdown_support( $post ) {
 	}
 	return $post;
 }
-add_filter( 'wp_insert_post_data', 'gutenberg_remove_wpcom_markdown_support', 9 );
+// add_filter( 'wp_insert_post_data', 'gutenberg_remove_wpcom_markdown_support', 9 ); tmp fix.
```

Comment:
What is the permanent fix here?

Reply:
Disabling Markdown relies on this check `gutenberg_content_has_blocks( $content )` which in turn is a simple `return false !== strpos( $content, '<!-- wp:' );`

The reason why it was implemented seems to be:

> WPCOM markdown support causes issues when saving a Gutenberg post by stripping out the `<p>` tags. This adds a filter prior to saving the post via REST API to disable markdown support. Fixes markdown support provided by plugins Jetpack, JP-Markdown, and WP Editor.

My solution was to implement that Gutenberg block protection in `WPCom_Markdown` by relying on the existing `jetpack_markdown_preserve_pattern` functionality which I believe was intended for this purpose: https://github.com/Automattic/jetpack/blob/master/_inc/lib/markdown/gfm.php#L106

I preserve all non Gutenberg non-Markdown blocks here:

https://github.com/Automattic/jetpack/pull/9357/commits/a1c661172672c2f0986c53ccc1bab1581f0edbdf#diff-c8a16334ab363043643dab428095f818R580

I do an equivalent of `gutenberg_content_has_blocks( $content )` to decide whether to protect the content or not.

I may need some feedback from @mkaz  and/or @aduth  on the viability of this solution.

I don't know the exact nature of the JP-Markdown, and WP Editor conflicts that are mentioned but I think this solution may scale to address those conflicts by simply adding more Markdown exclusion patterns like: exclude anything outside of the Markdown block. Perhaps that could be included now.